### DISCUSSION TITLE
Meerk40t grblcontrol lightburn integration issue

### DISCUSSION THREAD

**Discussion by Badrimus** [meerk40t/meerk40t#2374] [General]
This is my first post here,
I'm having issue integrating meerk40t to my lightburn (gcode Grbl version) 
I'm using Grbl LPC machine with localhost IP 127.0.0.1 , I telneted the localhost and it's connected but lightburn it's not able to control my k40 with M2 nano board after I send grblcontrol command from meerk40t console. 
I'm using Meerk40t latest version which controls my k40 but with a lot of lag over 2 seconds of lag sometimes more after trying to jog the laserhead for exemple.
Whisperer works fine.
Hopefully you guys can help me sort out this issue , I sure I'm missing something. Any help will be much appreciated.

**Comment by tatarize** [comment]
So the issue is there's a really strong amount of lag between meerk40t controlling your K40? Like no lightburn, no grblcontrol, you just have a 2 second delay controlling the laser? The lightburn and grblcontrol stuff will just add in some extra delay but it sounds like the lag is between meerk40t and the laser. And that lag does *not* exist in whisperer making it a defacto bug?

If MeerK40t and Whisperer act differently, the software is generally unrelated so it's always highlighting a bug if there's a defect that isn't on both (indicating hardware issues).

What's your log doing? If you go to controller and you view your USB log, does it say anything is going strangely or broken?

Secondly if you sent a raw `egv` command to the laser would that also lag? 
Type `egv IBzzzzS1P$` in console it would necessarily move the laser a bit over an inch to the right. 
When I type this and hit enter the laser is already moving by the time my finger comes back up. (Note: after doing this you will probably need to rehome the device since the program doesn't snoop on the packets to know you moved the laser (unless you want it to and `lhyinterpreter` in console would do that). -- In any event this basically injects the move command directly into the driver for sending.

**Comment by jpirnay** [comment]
Possibly outdated,
